frontend/src/components/ViewActivities/editActivity.js

```diff
@@ -96,17 +100,33 @@ class EditActivity extends React.Component {
    */
   countriesDropdown(ref, tzref) {
     return (
-      <Form.Control as="select" ref={ref} onChange={tzref}>
-        <option key="-1">No Change</option>
+      <Form.Control as='select' ref={ref} onChange={tzref}>
+        <option key='-1'>No Change</option>
         {countryList.map((item, index) => {
           return (
-            <option key={index} eventKey={index}>{item}</option>
+            <option key={index}>{item}</option>
           );
         })}
       </Form.Control>
     );
   }
 
+  /**
+   * Delete this activity. 
+   * 
+   * @returns true if the activity was successfully deleted.
+   */
+  async deleteActivity() {
+    if (window.confirm(`Are you sure you want to delete ${this.props.activity[DB.ACTIVITIES_TITLE]}?
+This action cannot be undone!`)) {
```

Comment:
Ahhhhhhhh this is so nice! I am definitely stealing this if that's cool with you.